query/src/main/java/io/keen/client/java/AbsoluteTimeframe.java

```diff
@@ -46,16 +46,20 @@ public AbsoluteTimeframe(String start, String end) {
      */
     @Override
     public Map<String, Object> constructTimeframeArgs() {
-        Map timeframe = null;
+        Map<String, Object> timeframe = null;
+
         if (start != null && end != null) {
-            Map absoluteTimeframe = new HashMap<String, Object>();
+            Map<String, Object> absoluteTimeframe = new HashMap<String, Object>(3);
```

Comment:
I'm fine with dropping support for Java 6. Official support for it ended a long time ago, and Android (the primary use case for this SDK) doesn't support it.